c/validator.h

```diff
@@ -72,13 +127,6 @@ void csal_change_init(csal_change_t *state, csal_entry_t *buffer,
 
 int csal_change_insert(csal_change_t *state, const uint8_t key[CSAL_KEY_BYTES],
                        const uint8_t value[CSAL_VALUE_BYTES]) {
-  if (state->length < state->capacity) {
```

Comment:
Can you explain a bit about the changes made here?

Reply:
There are some bugs in `csal_change_organize()`, I'll  try to add this optimize back.

Reply:
Just an FYI, the original code is already optimized to save some fiddling around when the VM is executing, I would suggest we keep this optimization, and try to tackle the bug in `csal_change_organize` alone.

Reply:
OK